pkg/cpuid/cpuid_tinygo_linux_amd64.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2012-2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build tinygo
+
+package cpuid
+
+/*
+#include <stdlib.h>
+char *CpuidVendor() {
+    unsigned int eax, ebx, ecx, edx;
+    char *vendor = (char *)(calloc(13, sizeof(char)));
+
+    // handle this in golang
+    if (vendor == NULL) {
+        return NULL;
+    }
+
+    eax = 0;
+    __asm__ volatile(
```

Comment:
ah gee, I just realized this code is good for both 386 and amd64. So  ... up to you ... if you want to put in the build tags instead of the file rename. Sorry.

Reply:
I agree, I don't think we will encounter any x86 32-bit systems in the wild very often. For me, it would be sufficient to keep it as is and neither add x86 to the file name nor att `386` to the build tags.